client/src/components/layout/TripForm.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { Redirect } from "react-router-dom";
+
+import translateServerErrors from "../../services/translateServerErrors.js"
+import ErrorList from "./ErrorList.js"
+
+
+const TripForm = (props) => {
+  const [form, setForm] = useState({
+    title: "",
+    continent: "",
+    country: "",
+    city: "",
+    numberOfDays: "",
+    description: ""
+  })
+
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [errors, setErrors] = useState([])
+
+  const addNewTrip = async () => {
+    try {
+      const response = await fetch("/api/v1/trips", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(form)
+      })
+      if (!response.ok) {
+        if (response.status == 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw(error)
+      } else {
+        const body = await response.json()
```

Comment:
If we are making the effort to parse the `response`, we should check the contents of the `body` from the response before we set redirect to true. Something like
```
const body = await response.json()
if (body.trip) {
  .....
```